public/app/features/templating/formatRegistry.ts

```diff
@@ -184,13 +190,30 @@ export const formatRegistry = new Registry<FormatRegistryItem>(() => {
       id: 'glob',
       name: 'Glob',
       description: 'Format multi valued variables using glob syntax, example {value1,value2}',
-      formatter: (value, args, variable) => {
+      formatter: ({ value }) => {
         if (isArray(value) && value.length > 1) {
           return '{' + value.join(',') + '}';
         }
         return value;
       },
     },
+    {
+      id: 'text',
+      name: 'Text',
+      description: 'Format variables in their text representation. Example in multi variable scenario A + B + C.',
+      formatter: ({ text }, variable) => {
+        if (typeof text === 'string') {
+          return text;
+        }
+
+        const current = (variable as any)?.current;
+
+        if (typeof current?.value === 'string') {
+          return current.value;
+        }
+        return current.text;
+      },
+    },
```

Comment:
nit: is this clearer?
```suggestion
       if(!current) {
         return '';
       }
       
       const {value, text} = current;

        if (typeof value === 'string') {
          return value;
        }
        return text;
      },
    },
```